gotests_test.go

```diff
@@ -918,7 +919,7 @@ func TestGenerateTests(t *testing.T) {
 				return
 			}
 			if got := string(gts[0].Output); got != tt.want {
-				t.Errorf("%q. GenerateTests(%v) = \n%v, want \n%v", tt.name, tt.args.srcPath, got, tt.want)
+				t.Errorf("%q. GenerateTests(%v) = diff=%v", tt.name, tt.args.srcPath, cmp.Diff(got, tt.want))
```

Comment:
Would it be possible to create maybe a custom reporter so we can get `got/want` with a lines of the diff instead `-/+` ?

also `%v` now can be `%s`

Reply:
The custom reporter approach certainly looks interesting. So far I just updated @mekegi's code, but I could give integrating the custom reporter a try.

So your goal would be to print the output of the custom reporter instead of the default cmp.Diff(x, y), right? It seems doable for this specific line of code, i.e. in the `gotests_test.go` it should be fairly trivial to add. 

Do you mean to also use the custom reporter for the generated tests? And by extension, making that configurable by the user of gotests too? Or am I way off and did you mean to just put it in `gotests_test.go`?

Reply:
Taking the example DiffReporter and changing the lines to this:

```go
var r reporter.DiffReporter
if got := string(gts[0].Output); !cmp.Equal(got, tt.want, cmp.Reporter(&r)) {
    t.Errorf("%q. GenerateTests(%v) = diff=%s", tt.name, tt.args.srcPath, r.String())
    outputResult(t, tmp, tt.name, gts[0].Output)
}

// basic Report() function, slightly altered from the example
func (r *DiffReporter) Report(rs cmp.Result) {
	if !rs.Equal() {
		vx, vy := r.path.Last().Values()
		r.diffs = append(r.diffs, fmt.Sprintf("\n\nGot:\n\n%+v\n\nWant:\n\n%+v\n", vx, vy))
	}
}
```

Results in the following output:

```bash
    gotests_test.go:924: "Receiver is indirect imported struct". GenerateTests(testdata/test037.go) = diff=

        Got:

        package testdata

        import "testing"

        func Test_someIndirectImportedStruct_Foo037(t *testing.T) {
        	tests := []struct {
        		name string
        		smtg *someIndirectImportedStruct
        	}{
        		// TODO: Add test cases.
        	}
        	for _, tt := range tests {
        		tt.smtg.Foo037()
        	}
        }


        Want:

        package testdata

        import "testing"

        func Test_someIndirectImportedStruct_Foo037(t *testing.T) {
        	tests := []struct {
        		name string
        		smtg *someIndirectImportedStruct
        	}{
        		// TODO: Add test cases.
        	}
        	for _, tt := range tests {
        		smtg := &someIndirectImportedStruct{}
        		smtg.Foo037()
        	}
        }

    gotests_test.go:958: /var/folders/m1/h8xf_67j5fqccwr1277f3_zc0000gn/T/gotests_test230212921/receiver_is_indirect_imported_struct.go
```

This doesn't seem really useful, but it may be possible to customize it down to the actual lines that are different. 

If I understand you correctly, you want to show just got/want and the corresponding lines, correct? I find the -/+ quite easy to use, so I'm not sure if we should drop that, unless you want it to become configurable for a user of gotests?

Reply:
I was talking only about generated tests and you are right without granular line detection it does not seem really useful.

Reply:
+1 to keeping the -/+. Without them it's very difficult to know what the delta is.